saltpack.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package updater
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+
+	"github.com/agl/ed25519"
+	"github.com/keybase/go-logging"
+	"github.com/keybase/saltpack"
+)
+
+type naclSigningKeyPublic [ed25519.PublicKeySize]byte
+type naclSignature [ed25519.SignatureSize]byte
+
+const kidNaclEddsa = 0x20
+
+// KID is a key identifier
+type KID string
+
+const (
+	// KIDLen is KID length in bytes
+	KIDLen = 35
+	// KIDSuffix is KID suffix (byte)
+	KIDSuffix = 0x0a
+	// KIDVersion is current version of KID
+	KIDVersion = 0x1
+)
```

Comment:
It looks like we have to duplicate these constants from libkb, because we specify the set of valid libkb-style KIDs, rather than just using a set of raw signing public keys. Any reason not to just use `"9092ae4e790763dc7343851b977930f35b16cf43ab0ad900a2af3d3ad5cea1a1"` instead of `"01209092ae4e790763dc7343851b977930f35b16cf43ab0ad900a2af3d3ad5cea1a10a"`, and avoid the KID-munging code?